tests/pytests/integration/modules/test_pillar.py

```diff
@@ -478,3 +478,118 @@ def test_pillar_refresh_pillar_scheduler(salt_cli, salt_minion):
 
     # Refresh pillar once we're done
     salt_cli.run("saltutil.refresh_pillar", wait=True, minion_tgt=salt_minion.id)
+
+
+@pytest.mark.slow_test
+def test_pillar_match_filter_by_minion_id(salt_cli, salt_minion):
```

Comment:
@saltstack/team-core Let's please create a followup task. These are actually 3 tests which can be parametrized with pytest.

Reply:
Sorry, I suck at tests 😄

Reply:
No worries, pytest is also recent in Salt :grinning:

Reply:
I'll tackle this probably tomorrow.

Reply:
Here we go: https://github.com/saltstack/salt/pull/59889